x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/fleet_server_callouts/fleet_server_missing_encryption_key_callout.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { MouseEventHandler } from 'react';
+import { EuiButtonEmpty, EuiCallOut, EuiLink } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { useStartServices } from '../../../../hooks';
+
+export interface FleetServerMissingEncryptionKeyCalloutProps {
+  onClickHandler: MouseEventHandler<HTMLButtonElement>;
+}
+
+export const FleetServerMissingEncryptionKeyCallout: React.FunctionComponent<
+  FleetServerMissingEncryptionKeyCalloutProps
+> = ({ onClickHandler }) => {
+  const { docLinks } = useStartServices();
+  return (
+    <EuiCallOut
+      data-test-subj="missingEncryptionKeyCallout"
+      iconType="iInCircle"
+      color="warning"
+      title={
+        <FormattedMessage
+          id="xpack.fleet.missingEncryptionKeyCallout.title"
+          defaultMessage="Set up encryption key"
+        />
+      }
+    >
+      <p data-test-subj="missingEncryptionKeyBody">
+        <FormattedMessage
+          id="xpack.fleet.missingEncryptionKeyCallout.message"
+          defaultMessage="An encryption key will make your environment more secure. Click {link} to learn how to set up an encryption key."
+          values={{
+            link: (
+              <EuiLink href={docLinks.links.kibana.secureSavedObject} target="_blank">
+                <FormattedMessage
+                  id="xpack.fleet.missingEncryptionKeyCallout.helpLink"
+                  defaultMessage="here"
+                />
+              </EuiLink>
+            ),
+          }}
+        />
+      </p>
+      <EuiButtonEmpty flush="left" size="s" color="warning" onClick={onClickHandler}>
```

Comment:
Correct me if I'm wrong, but I guess we are sending the `click ev` as a value here, can we call the function with a boolean value? so we don't store the entire `ev` at local storage. 

```typescript
<EuiButtonEmpty flush="left" size="s" color="warning" onClick={() => {onClickHandler(true);}}>
```

Or at the parent component. 
Also, we could memoize that function.